includes/plugins/class-wc-memberships.php

```diff
@@ -335,6 +390,40 @@ public static function excerpt( $excerpt, $post, $message_code ) {
 		return $excerpt;
 	}
 
+	/**
+	 * Render the overlay gate.
+	 */
+	public static function render_overlay_gate() {
+		if ( ! self::has_gate() ) {
+			return;
+		}
+		// Only render overlay gate for a restricted singular content.
+		if ( ! is_singular() || ! self::is_post_restricted() ) {
+			return;
+		}
+		$gate_post_id = self::get_gate_post_id();
+		$style        = \get_post_meta( $gate_post_id, 'style', true );
+		if ( 'overlay' !== $style ) {
+			return;
+		}
+		global $post;
+		$_post = $post;
+		$post  = \get_post( $gate_post_id ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
```

Comment:
Overwriting the global seems risky—any particular reason why we need this?

Reply:
Some `the_content` filter is forcefully replacing the gate `post_content` with the global post's content. This was the only way I could make the gate content be rendered properly. The global variable is immediately restored to its original value after the gate is rendered, so I don't think we are doing anything bad.

You can test it locally by commenting-out lines 409-414 and 423-424 and checking if the `post_content` is working.

If your local instance manages to render the correct content, it's probably something here that I should figure out.

Reply:
I'm seeing the same thing, and I think it's the `apply_filters( 'the_content' )` that's messing with the post returning the content, but it's also necessary to transform the block markup correctly, so let's leave it as-is.